### ISSUE TITLE
LookupError with long Windows usernames

### ISSUE THREAD

**Issue by timlehr** [pypa/setuptools-scm#205]
Hey guys,

a quick bug report before we all enjoy our holidays. :)

I ran into trouble with the library, as it will an LookupError when the username is truncated by Windows with "~1" for being too long. Or at least that's what I think what's causing it. On another user named 'tlehr' everything works as expected. Any clues on how to fix this? With 1.15.0 everything works even with a long username. Any newer version breaks. This was tested on Windows 10.

Cheers,
Tim
```
Processing c:\users\lgotkowski\scarif
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\lgotko~1\appdata\local\temp\pip-jmjozo-build\setup.py", line 31, in <module>
        dependency_links=_dep_links
      File "C:\Python27\Lib\distutils\core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "c:\users\lgotkowski\scarif\env\lib\site-packages\setuptools\dist.py", line 318, in __init__
        _Distribution.__init__(self, attrs)
      File "C:\Python27\Lib\distutils\dist.py", line 287, in __init__
        self.finalize_options()
      File "c:\users\lgotkowski\scarif\env\lib\site-packages\setuptools\dist.py", line 376, in finalize_options
        ep.load()(self, ep.name, value)
      File "c:\users\lgotko~1\appdata\local\temp\pip-jmjozo-build\.eggs\setuptools_scm-1.15.6-py2.7.egg\setuptools_scm\integration.py", line 22, in version_keyword
      File "c:\users\lgotko~1\appdata\local\temp\pip-jmjozo-build\.eggs\setuptools_scm-1.15.6-py2.7.egg\setuptools_scm\__init__.py", line 119, in get_version
      File "c:\users\lgotko~1\appdata\local\temp\pip-jmjozo-build\.eggs\setuptools_scm-1.15.6-py2.7.egg\setuptools_scm\__init__.py", line 97, in _do_parse
    LookupError: setuptools-scm was unable to detect version for 'c:\\users\\lgotko~1\\appdata\\local\\temp\\pip-jmjozo-build'.

    Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

    For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj

    ----------------------------------------

Error: Command "python setup.py egg_info" failed with error code 1 in c:\users\lgotko~1\appdata\local\temp\pip-jmjozo-build\
```

**Comment by woodyza** [comment]
Hey,

I had a similar issue recently, have you tried installing from master instead? #174 fixes an issue with the crusty old Windows 8.3 filename truncation but it's not yet rolled into an official release.

**Comment by timlehr** [comment]
Hi woodyza,

Thanks for the quick reply!

No I haven’t yet, but #174  seems like the issue I‘m experiencing.
I hope there will be a new release for PIP soon, as my deployment system is built around it.

**Comment by jaraco** [comment]
I find this to be a really interesting issue. I was surprised to see MS-DOS names still causing problems. I thought surely if I pulled out a brand-new Windows 10 Fall Creators Update that it wouldn't be generating short file names or using them by default, but in fact it is (yuck!).

I did some more digging, thinking that maybe Python wasn't returning the canonical name in the `tempfile` module, but no - even calling `GetTempPath` API returns the short name (but [only in the C++ API and not .Net](https://social.msdn.microsoft.com/Forums/silverlight/en-US/a9372c59-763c-4a3c-9d0f-81283347ae7a/c-gettemppath-different-than-c-pathgettemppath?forum=vcgeneral)) (what a mess!).

 I then discovered that it's possible to [disable short file names](https://support.microsoft.com/en-us/help/121007/how-to-disable-8-3-file-name-creation-on-ntfs-partitions) for a given volume. I invoked that command to disable the 8.3 file name creation, created a new account with a long name, and then verified that the temp directory is created without the munged name. Cool, but little help for a home directory that was previously created with the 8.3 creation enabled, because the setting only affects directories created after the setting is set.

Investigating further, I found it's [possible to disable use of that 8.3 name for all subdirectories](https://www.experts-exchange.com/questions/28501555/How-do-I-remove-existing-short-filenames-from-Windows-2008-Server-not-R2.html) of a select directory. So I restored the default behavior (creation of 8.3 names), recreated the longusername account, and then using an administrator account, stripped the 8.3 name:

```
PS C:\Windows\system32> fsutil 8dot3name strip /v C:\users
Scanning registry...
Registry Data                                                                     Registry Key Path
-------------------------------------------------------------------------------   ------------------------------------------

Total affected registry keys:                   0

Stripping 8dot3 names...

8dot3 Name      FileId                Full Path
-------------   -------------------   -------------------------------------------------------------------------------------
ALLUSE~1        0x1000000007774       "C:\users\All Users"
DEFAUL~1        0x1000000007775       "C:\users\Default User"
LONGUS~1        0x8000000007861       "C:\users\longusername"

Total files and directories scanned:            7
Total 8dot3 names found:                        3
Total 8dot3 names stripped:                     3

For details on the operations performed please see the log:
  "C:\Users\jaraco\AppData\Local\Temp\8dot3_removal_log @(GMT 2017-12-25 18-02-17).log"
```

Annoyingly, it seems it's not possible to invoke that on only one directory (such as `c:\Users\longusername`). If you invoke it on `C:\Users\longusername`, it alters the subdirectories of that indicated directory. Bad design, but whatever.

After doing that, and logging into the user's account once again, the 8.3 name no longer appears:

```
PS C:\Users\longusername> py
Python 3.6.3 (v3.6.3:2c5fed8, Oct  3 2017, 18:11:49) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tempfile
>>> tempfile.gettempdir()
'C:\\Users\\longusername\\AppData\\Local\\Temp'
```

This approach feels like a more proper fix as it will eliminate the alias and cause all operations in all applications to use the canonical name, and not just this one use case in setuptools_scm. I present it here for your information and as a possible workaround. Do use with some caution, as there appears to be no reverse operation for `fsutil 8dot3name strip`.
